common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -54,12 +54,21 @@ def get_queryset(self):
         """
         if self.requested_enterprise_uuid is None:
             raise ParseError('Required enterprise_customer_uuid is missing')
-        enterprise_customer_idp = get_object_or_404(
+        enterprise_customer_idps = get_list_or_404(
             EnterpriseCustomerIdentityProvider,
             enterprise_customer__uuid=self.requested_enterprise_uuid
         )
-        return SAMLProviderConfig.objects.current_set().filter(
-            slug=convert_saml_slug_provider_id(enterprise_customer_idp.provider_id))
+        slug_list = []
+        config_list = []
+        for idp in enterprise_customer_idps:
+            slug_list.append(convert_saml_slug_provider_id(idp.provider_id))
+
+        for config in SAMLProviderConfig.objects.current_set():
```

Comment:
and then I think you can do -
```
config_list = [ config in SAMLProviderConfig.objects.current_set() if convert_saml_slug_provider_id(config.provider_id) in slug_test ]
```

Reply:
hmm actually I'm seeing that we're converting `enterprise_customer_idps` for the slug list and then converting the current set from the `SAMLProviderConfig.objects` in the same way. Could we just compare both of those values directly without needing to convert?